compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1983,39 +1983,10 @@ private void checkRecordVarRefEquivalency(Location pos, BLangRecordVarRef lhsVar
                                               Location rhsPos) {
         rhsType = types.getReferredType(rhsType);
         if (rhsType.tag == TypeTags.MAP) {
-            BMapType rhsMapType = (BMapType) rhsType;
-            BType expectedType;
-            switch (rhsMapType.constraint.tag) {
-                case TypeTags.ANY:
-                case TypeTags.ANYDATA:
-                case TypeTags.JSON:
-                    expectedType = rhsMapType.constraint;
-                    break;
-                case TypeTags.UNION:
-                    BUnionType unionType = (BUnionType) rhsMapType.constraint;
-                    LinkedHashSet<BType> unionMemberTypes = new LinkedHashSet<BType>() {{
-                        addAll(unionType.getMemberTypes());
-                        add(symTable.nilType);
-                    }};
-                    expectedType = BUnionType.create(null, unionMemberTypes);
-                    break;
-                default:
-                    expectedType = BUnionType.create(null, new LinkedHashSet<BType>() {{
-                        add(rhsMapType.constraint);
-                        add(symTable.nilType);
-                    }});
-                    break;
+            for (BLangRecordVarRefKeyValue field: lhsVarRef.recordRefFields) {
+                dlog.error(field.variableName.pos,
+                        DiagnosticErrorCode.INVALID_FIELD_BINDING_PATTERN_WITH_NON_REQUIRED_FIELD);
```

Comment:
We can improve the error message including the field name. Create an issue for this.

Reply:
Created and issue https://github.com/ballerina-platform/ballerina-lang/issues/34375